src/bun.js/child_process.exports.js

```diff
@@ -1,3 +1,995 @@
+ * @param {number} [minLength]
+ * @returns {asserts value is any[]}
+ */
+
+/** @type {validateArray} */
+const validateArray = (value, name, minLength = 0) => {
+  // const validateArray = hideStackFrames((value, name, minLength = 0) => {
+  if (!ArrayIsArray(value)) {
+    throw new ERR_INVALID_ARG_TYPE(name, "Array", value);
+  }
+  if (value.length < minLength) {
+    const reason = `must be longer than ${minLength}`;
+    throw new ERR_INVALID_ARG_VALUE(name, value, reason);
+  }
+};
+
+/**
+ * @callback validateString
+ * @param {*} value
+ * @param {string} name
+ * @returns {asserts value is string}
+ */
+
+/** @type {validateString} */
+function validateString(value, name) {
+  if (typeof value !== "string")
+    throw new ERR_INVALID_ARG_TYPE(name, "string", value);
+}
+
+/**
+ * @param {?object} options
+ * @param {string} key
+ * @param {boolean} defaultValue
+ * @returns {boolean}
+ */
+function getOwnPropertyValueOrDefault(options, key, defaultValue) {
+  return options == null || !ObjectPrototypeHasOwnProperty(options, key)
+    ? defaultValue
+    : options[key];
+}
+
+function ObjectPrototypeHasOwnProperty(obj, prop) {
+  return Object.prototype.hasOwnProperty.call(obj, prop);
+}
+
+function ArrayPrototypePush(array, ...items) {
+  return array.push(...items);
+}
+
+function ArrayPrototypeReduce(array, callback, initialValue) {
+  return array.reduce(callback, initialValue);
+}
+
+function ArrayPrototypeJoin(array, separator) {
+  return array.join(separator);
+}
+
+function ArrayPrototypeMap(array, callback) {
+  return array.map(callback);
+}
+
+function ArrayPrototypeSlice(array, begin, end) {
+  return array.slice(begin, end);
+}
+
+function ArrayPrototypeFilter(array, callback) {
+  return array.filter(callback);
+}
+
+function ArrayPrototypeIncludes(array, searchElement, fromIndex) {
+  return array.includes(searchElement, fromIndex);
+}
+
+function ArrayIsArray(arg) {
+  return Array.isArray(arg);
+}
+
+function isArrayBufferView(value) {
+  return value instanceof ArrayBufferView;
+}
+
+//------------------------------------------------------------------------------
+// 6. Node error polyfills
+//------------------------------------------------------------------------------
+
+function ERR_INVALID_ARG_TYPE(name, type, value) {
+  return new Error(
+    `The argument '${name}' is invalid. Received '${value}' for type '${type}'`
+  );
+}
+
+function ERR_INVALID_ARG_VALUE(name, value, reason) {
+  return new Error(
+    `The value '${value}' is invalid for argument '${name}'. Reason: ${reason}`
+  );
+}
+
+// TODO: Add actual proper error implementation here
+function errnoException(err, name) {
+  return new Error(`Error: ${name}. Internal error: ${err.message}`);
+}
+
+//------------------------------------------------------------------------------
+// 7. Node stream polyfills
+//------------------------------------------------------------------------------
+Readable.prototype.on = function (event, listener) {
```

Comment:
This overrides `Readable.prototype.on` globally, which we should not do. 

Is this a workaround for a bug in `Readable` in Bun? if so, can you file an issue for that and possibly fix it? (& verify the behavior matches Node)

If it isn't and this is expected behavior specific to `ChildProcess`, can we return a subclass the `Readable` objects returned by `ChildProcess` that does this? 

 Instead, please use a subclass we extend from. 

If we need to override it from user input,